### PR TITLE
40927 floating menu bug, take two

### PR DESCRIPTION
**Pull request by peterwilsoncc**
https://core.trac.wordpress.org/ticket/40927

### CODE REVIEW

File: tests/phpunit/tests/admin/includesPlugin.php [hunk 1]
```diff
@@ -257,7 +257,13 @@ public function data_submenu_position() {
 			array( -1, 0 ),                    // Negative numbers are treated the same as passing 0.
 			array( -7, 0 ),                    // Negative numbers are treated the same as passing 0.
 			array( 1, 1 ),                     // Insert as the second item.
+			array( '1', 1 ),                   // Insert as the second item.
+			array( 1.5, 1 ),                   // Insert as the second item.
+			array( '1.5', 1 ),                 // Insert as the second item.
 			array( 3, 3 ),                     // Insert as the 4th item.
+			array( '3', 3 ),                   // Insert as the 4th item.
+			array( 3.5, 3 ),                   // Insert as the 4th item.
+			array( '3.5', 3 ),                 // Insert as the 4th item.
```
Comment (sybrew) [inline tests/phpunit/tests/admin/includesPlugin.php:268]: Also try testing for `(string) "42e0"` and other values that return `true` in `is_numeric()`.

Comment (peterwilsoncc) [reply tests/phpunit/tests/admin/includesPlugin.php:268]: > Also try testing for `(string) "42e0"` and other values that return `true` in `is_numeric()`.

Good call, added `3e0` in e0ceb288b4f4d4f472d0a260f0c06e8c3ace3f08

Unfortunately PHP broke backcompat for items some with whitespace in 8.0 (eg ` 100 `) so I've left them alone rather than make the tests conditional for an edge case.

Comment (peterwilsoncc) [reply tests/phpunit/tests/admin/includesPlugin.php:268]: The white space was helpfully trimmed in my example `[space]100[space]`.

**Comment by peterwilsoncc** [comment]
https://core.trac.wordpress.org/changeset/53104
